modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/CommerceOrderHttpHelperImpl.java

```diff
@@ -588,6 +599,97 @@ public boolean hasCommerceOrderPortlet(
 		return false;
 	}
 
+	@Override
+	public boolean hasReturns(HttpServletRequest httpServletRequest) {
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-10562")) {
+			return false;
+		}
+
+		CommerceOrder commerceOrder =
+			CommerceOrderInfoItemUtil.getCommerceOrder(
+				_commerceOrderService, httpServletRequest);
+
+		if ((commerceOrder != null) && !commerceOrder.isOpen()) {
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.
+					fetchObjectDefinitionByExternalReferenceCode(
+						"L_COMMERCE_RETURN", CompanyThreadLocal.getCompanyId());
+
+			if (objectDefinition == null) {
+				return false;
+			}
+
+			return ListUtil.exists(
+				_objectEntryLocalService.getObjectEntries(
+					0, objectDefinition.getObjectDefinitionId(),
+					QueryUtil.ALL_POS, QueryUtil.ALL_POS),
+				objectEntry -> {
+					Map<String, Serializable> values = objectEntry.getValues();
+
+					if (commerceOrder.getCommerceOrderId() ==
+							GetterUtil.getLong(
+								values.get(
+									"r_commerceOrderToCommerceReturns" +
+										"_commerceOrderId"))) {
+
+						return true;
+					}
+
+					return false;
+				});
+		}
+
+		return false;
+	}
+
+	@Override
+	public boolean hasShipments(HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		CommerceOrder commerceOrder =
+			CommerceOrderInfoItemUtil.getCommerceOrder(
+				_commerceOrderService, httpServletRequest);
+
+		if ((commerceOrder != null) && !commerceOrder.isOpen() &&
+			_commerceOrderModelResourcePermission.contains(
+				PermissionThreadLocal.getPermissionChecker(), commerceOrder,
+				ActionKeys.VIEW)) {
+
+			List<CommerceShipment> commerceShipment =
+				_commerceShipmentLocalService.getCommerceShipments(
+					commerceOrder.getCommerceOrderId(), 0, 1);
+
+			if (!commerceShipment.isEmpty()) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	@Override
+	public boolean isCommerceOrderOpen(HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		CommerceOrder commerceOrder =
+			CommerceOrderInfoItemUtil.getCommerceOrder(
+				_commerceOrderService, httpServletRequest);
+
+		if ((commerceOrder == null) ||
```

Comment:
This check doesn't look right: this is boolean method in which you're gonna say that an order is *closed* (not open) if a user doesn't have VIEW permissions to access it. But then it means that those who'll use this method won't know what `isCommerceOrderOpen` is about. Please remove this `if`, permission checking is already [handled](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-api/src/main/java/com/liferay/commerce/util/CommerceOrderInfoItemUtil.java#L49).